the-collab-lab/tcl-72-smart-shopping-list
# 11. As an item, I want my estimated next purchase date to be computed at the time my purchase is recorded in the database so the app can learn how often I buy different items

## Description
New feature



## Related Issue


Closes #11 

## Acceptance Criteria



## Type of Changes

Enhancement

## Updates
When user purchases an item, the item data is updated on firestore, specifically the dateLastPurchased and dateNextPurchased property and estimates how often the user is going to purchase the item in the future.

### Before
User purchases an item, only dateLastPurchased and totalPurchases property is changed.


### After
User purchases an item, dateLastPurchased, dateNextPurchased, and totalPurchases is updated. 
dateNextPurchased is calculated by estimating how long until the user needs to purchase the item again using the function calculateEstimate.



## Testing Steps / QA Criteria
1. switch to branch kn-ho-next-purchase
2. If you do not already have an item on your shopping list, then add an item on the ManageList page.
3. Go to the List page and it should render the items in your shopping list.
4. Click on the checkbox to simulate a user buying an item.
5. Go to Firestore to check the dateLastPurchased, dateNextPurchased, and totalPurchases property.
6. To check if dateNextPurchased is properly calculated, manipulate the dateLastPurchased and totalPurchases property. For example:
  - If totalPurchases > 2 and dateLastPurchased = 7 days after dateCreated, then dateNextPurchased should also be 7 days after dateLastPurchased
